### ISSUE TITLE
Crash in -[MGLMapView semanticContentAttribute] on iOS 8

### ISSUE THREAD

**Issue by georgejlee** [mapbox/mapbox-gl-native#10225]
**Platform:** iOS 8.4
**Mapbox SDK version:** 3.6.4

Tested on iPhone 5 simulator with iOS 8.4 on Xcode 9.

### Steps to trigger behavior

 1. Open app
 2. Pinch to zoom
 3. Crash

### Expected behavior

Should not crash when zooming.

### Actual behavior

Crash in `-[MGLMapView semanticContentAttribute]` (see trace below). Possibly related, I am seeing crashes in `-[MGLScaleBar usesRightToLeftLayout] (MGLScaleBar.mm:181)` on iOS 8 devices in Crashlytics. Is iOS 8 still supported? The documentation at https://github.com/mapbox/mapbox-gl-native/tree/master/platform/ios says that it works with iOS 8.0 and above.

```
2017-10-18 09:52:43.966 PastureMap[78768:31321727] -[MGLMapView semanticContentAttribute]: unrecognized selector sent to instance 0x7a2d0600
2017-10-18 09:52:43.969 PastureMap[78768:31321727] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[MGLMapView semanticContentAttribute]: unrecognized selector sent to instance 0x7a2d0600'
*** First throw call stack:
(
	0   CoreFoundation                      0x07807746 __exceptionPreprocess + 182
	1   libobjc.A.dylib                     0x06fd9a97 objc_exception_throw + 44
	2   CoreFoundation                      0x0780f705 -[NSObject(NSObject) doesNotRecognizeSelector:] + 277
	3   CoreFoundation                      0x07756287 ___forwarding___ + 1047
	4   CoreFoundation                      0x07755e4e _CF_forwarding_prep_0 + 14
	5   Mapbox                              0x027224ee _Z30MGLTileSetFromTileURLTemplatesP7NSArrayIP8NSStringEP12NSDictionaryIS1_P11objc_objectE + 5162
	6   Mapbox                              0x0272238c _Z30MGLTileSetFromTileURLTemplatesP7NSArrayIP8NSStringEP12NSDictionaryIS1_P11objc_objectE + 4808
	7   Mapbox                              0x02722897 _Z30MGLTileSetFromTileURLTemplatesP7NSArrayIP8NSStringEP12NSDictionaryIS1_P11objc_objectE + 6099
	8   Mapbox                              0x02722717 _Z30MGLTileSetFromTileURLTemplatesP7NSArrayIP8NSStringEP12NSDictionaryIS1_P11objc_objectE + 5715
	9   Mapbox                              0x026fdb1a _Z31MGLGeoJSONOptionsFromDictionaryP12NSDictionaryIP8NSStringP11objc_objectE + 472592
	10  Mapbox                              0x026e6215 _Z31MGLGeoJSONOptionsFromDictionaryP12NSDictionaryIP8NSStringP11objc_objectE + 376075
	11  UIKit                               0x0522b4f7 _UIGestureRecognizerSendActions + 327
	12  UIKit                               0x05229d74 -[UIGestureRecognizer _updateGestureWithEvent:buttonEvent:] + 561
	13  UIKit                               0x0522bdbd -[UIGestureRecognizer _delayedUpdateGesture] + 60
	14  UIKit                               0x0522f73a ___UIGestureRecognizerUpdate_block_invoke661 + 57
	15  UIKit                               0x0522f5fd _UIGestureRecognizerRemoveObjectsFromArrayAndApplyBlocks + 317
	16  UIKit                               0x05223506 _UIGestureRecognizerUpdate + 3720
	17  UIKit                               0x04e4390b -[UIWindow _sendGesturesForEvent:] + 1356
	18  UIKit                               0x04e44770 -[UIWindow sendEvent:] + 770
	19  UIKit                               0x04e02681 -[UIApplication sendEvent:] + 242
	20  UIKit                               0x04e12ab8 _UIApplicationHandleEventFromQueueEvent + 21484
	21  UIKit                               0x04de62e7 _UIApplicationHandleEventQueue + 2300
	22  CoreFoundation                      0x0772906f __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 15
	23  CoreFoundation                      0x0771eb7d __CFRunLoopDoSources0 + 253
	24  CoreFoundation                      0x0771e0d8 __CFRunLoopRun + 952
	25  CoreFoundation                      0x0771da5b CFRunLoopRunSpecific + 443
	26  CoreFoundation                      0x0771d88b CFRunLoopRunInMode + 123
	27  GraphicsServices                    0x0bb452c9 GSEventRunModal + 192
	28  GraphicsServices                    0x0bb45106 GSEventRun + 104
	29  UIKit                               0x04dea0b6 UIApplicationMain + 1526
	30  PastureMap                          0x0030678b main + 75
	31  libdyld.dylib                       0x088bbac9 start + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

![screen shot 2017-10-18 at 9 53 05 am](https://user-images.githubusercontent.com/135481/31731667-9f4a539a-b3ea-11e7-8888-164f860bf8a2.png)

**Comment by friedbunny** [comment]
Thanks for the report. iOS 8 is the minimum version we support, but occasionally incompatible features will accidentally slip in. 

In this case, enabling the scale bar is what’s causing the crash — if you need an immediate work-around, I’d suggest disabling the scale bar for iOS 8 users.

**Comment by friedbunny** [comment]
Fixed in #10241; will be in v3.7.0-beta.3.
